packages/haul-core/src/server/launchBrowser.ts

```diff
@@ -28,8 +28,17 @@ export default function launchBrowser(runtime: Runtime, url: string) {
     {
       // No Canary / Chrome, let's run Safari
       darwin: openWith('safari'),
-    }
-  ).catch((error: Error) => {
+    },
+  ];
+
+  // If the user has provided a specific browser preference, try that one first.
```

Comment:
There's certainly an argument to be made for **not** doing a fallback if the user has specified a preference.